app/models/student.rb

```diff
@@ -141,45 +138,33 @@ def create_group_for_working_alone_student(aid)
         end
       end
 
-      @grouping.group = @group
-      begin
-        unless @grouping.save
-          m_logger = MarkusLogger.instance
-          m_logger.log("Could not create a grouping for Student '#{user_name}'"\
+      empty_grouping = Grouping.find_by(assignment_id: aid, group_id: @group.id)
```

Comment:
I looked at the logic and the old issue. This seems unnecessary (at least now, maybe useful before). If a grouping already exists, it should still be the case that we can add the student to the grouping without destroying and recreating it.

Reply:
There is a unique index in the database on (assignement_id + group_id) in the groupings table.
I was able to reproduce the issue: you create your group as a student, then log in as the instructor and remove the student membership from the group (without deleting the group), then try to create it again as a student.

Reply:
@adisandro weren't you going to replace this with a `find_or_create_by`?

Reply:
Shoot I confused this pull request with the other about git.